In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import statsmodels
from statsmodels.stats.proportion import proportions_ztest

### Problem Statement :

Fantaloons Sales managers commented that % of males versus females walking in to the store differ based on day of the week. Analyze the data and determine whether there is evidence at 5 % significance level to support this hypothesiss

### 1. Business Issue  
Does the % of males versus females walking in to the store differ based on day of the week ?

### 2. Data description

α == 0.05 (95% Confidence)

Y == Discrete  
X == Discrete  

Since there are 2 variable we will perform 2 proportion test

### 3. Proportion Test

In [2]:
df = pd.read_csv("Faltoons.csv")

H0 == The % of males versus females walking in to the store based on day of the week is equal  
H1 == The % of males versus females walking in to the store based on day of the week is not equal

In [3]:
df.Weekend.str.count("Male").sum() # 167

167

In [4]:
df.Weekend.str.count("Female").sum() # 233

233

In [5]:
df.Weekdays.str.count("Male").sum() # 133

113

In [6]:
df.Weekdays.str.count("Female").sum() # 287

287

In [7]:
female_count = np.array([287, 233])

In [8]:
total_count = np.array([420,400])

In [9]:
statsmodels.stats.proportion.proportions_ztest(female_count, total_count)

(2.9964248640358, 0.0027316554639882896)

(2.9964248640358, 0.0027316554639882896)  
P value of 2 Propotion test  == 0.0027 < α

## Hence H1 is accepted.  
## Percentage of males vs females walking in to the store based on the day of the week is not similar

H0 == The % of females walking in to the store based on day of the week > then % of males  
H1 == The % of males walking in to the store based on day of the week < then % of males

In [10]:
statsmodels.stats.proportion.proportions_ztest(female_count, total_count, alternative = 'smaller')

(2.9964248640358, 0.9986341722680059)

(2.9964248640358, 0.9986341722680059)  
P value of 2 Propotion Test == 0.99 > α  

## Hence H0 is accepted.  
## Percentage of females walking in to the store based on the day of the week is > then Percentage of males